test/zIntegrationTests.ts

```diff
@@ -127,3 +162,76 @@ describe("Test suit for setting up the stage", function () {
         expect(await r.leverageAllocator.getAddressToLvUSDAvailable(await user.getAddress())).to.equal(parseUnitsNum(initialUserLevAllocation));
     });
 });
+
+describe("Test suit for getting leverage", function () {
+    let leverageUserIsTaking: number;
+
+    before(async function () {
+        await setupEnvForIntegrationTests();
+        leverageUserIsTaking = getFloatFromBigNum(
+            await r.parameterStore.getAllowedLeverageForPosition(userOUSDPrincipleInEighteenDecimal, numberOfCycles));
+        console.log("Will take %s lev while user has allocation of %s", leverageUserIsTaking,
+            getFloatFromBigNum(await r.leverageAllocator.getAddressToLvUSDAvailable(user.address)));
+
+        await approveAndGetLeverageAsUser(userOUSDPrincipleInEighteenDecimal, numberOfCycles, r, user);
+        positionId = 0;
+    });
+
+    it("Should have created a single position and assign it to user", async function () {
+        printPositionState(r, positionId);
+        printPoolState(r.curveLvUSDPool);
+        printMiscInfo(r, user);
+        const nftBalance = await r.positionToken.balanceOf(user.address);
+        expect(nftBalance).to.equal(1);
+    });
+
+    it("Should have assigned origination fee to treasury", async function () {
+        const treasuryBalance = getFloatFromBigNum(
+            await r.externalOUSD.balanceOf(r.treasurySigner.address),
+        );
+        /// origination fee is 5% at the moment, lvUSDBorrowed is 171 so 5% of it is roughly 8.5
```

Comment:
eventually would be good to make this calculation use variables in case we end up changing the fee or lvUSDBorrowed amounts in the future

Reply:
Agreed. its a lower priority at the moment